feat: add no-show hosts and csat graphs on Insights

hackice20 opened the pull request:
## What does this PR do?

This PR introduces two new time-series graphs to the insights page: "No-Show Hosts over Time" and "CSAT over Time". It also refactors the existing KPI cards and new charts to be displayed as individual, styled cards with curved edges and a subtle elevation, improving the overall UI/UX of the insights page to match the design system.

- Fixes #24532
- Fixes Linear `CAL-6600` (from the original issue)

## Visual Demo

I will provide screenshots for the visual demo section. For now, here are placeholders.

#### Before:
<img width="3072" height="1728" alt="image" src="https://github.com/user-attachments/assets/f16f3fe9-343a-4e7e-b141-be71df356b82" />


#### After:

<img width="3072" height="1728" alt="Screenshot From 2025-10-20 09-39-36" src="https://github.com/user-attachments/assets/9c880807-acb2-4521-afd1-33d734fab49c" />
<img width="3072" height="1728" alt="Screenshot From 2025-10-20 09-39-26" src="https://github.com/user-attachments/assets/6e953339-1201-482f-954e-ed2be9070600" />
<img width="3072" height="1728" alt="Screenshot From 2025-10-20 09-39-17" src="https://github.com/user-attachments/assets/ecf212de-d979-42d9-92d5-21e1f3c2fb8c" />
<img width="3072" height="1728" alt="Screenshot From 2025-10-20 09-38-54" src="https://github.com/user-attachments/assets/cd64cbae-72a5-4709-84e7-017d0f6adab8" />


## Mandatory Tasks

- [x] I have self-reviewed the code.
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). (N/A)
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works. (No new tests were added, as this PR follows existing patterns.)

## How should this be tested?

-   **Environment variables**: No special environment variables are needed.
-   **Test data**: The database should contain some bookings, including some with `noShowHost = true` and some with ratings to properly test the charts.

**Steps to test:**

1.  Navigate to the `/insights` page.
2.  Confirm that the "Events" and "Performance" sections now display each metric in a separate, styled card.
3.  Confirm that two new charts, "No-Show Hosts over Time" and "CSAT over Time," appear below the "Event Trends" chart, also in their own cards.
4.  Interact with the date filter to ensure the charts update correctly.
5.  Hover over the data points on the charts to verify that the tooltips display the correct information.
6.  Check that the layout is responsive on different screen sizes.

## Checklist

- My code follows the style guidelines of this project.
- I have commented my code, particularly in hard-to-understand areas.
- I have checked if my changes generate no new warnings.

***
    
<!-- This is an auto-generated description by cubic. -->
---

## Summary by cubic
Adds two new time-series graphs—No-Show Hosts over Time and CSAT over Time—and updates insights UI with individual, styled cards for clarity. Aligns with Linear CAL-6600 and closes #24532.

- **New Features**
  - Added NoShowHostsOverTimeChart and CSATOverTimeChart with tooltips and date-range filtering.
  - Implemented TRPC endpoints and service methods to aggregate no-show host counts and compute CSAT percentages.
  - Integrated the new charts and updated RecentNoShowGuests into the insights page layout.

- **Refactors**
  - Converted KPI sections into separate rounded cards with subtle elevation and consistent spacing.

<!-- End of auto-generated description by cubic. -->

hackice20 committed:
feat: Add No-Show Hosts and CSAT insights with beautiful card design

- Add No-Show Hosts Over Time chart with line visualization
- Add CSAT Over Time chart with percentage tracking
- Add Recent No-Show Guests list with email copy functionality
- Redesign all KPI cards as individual separated cards with curved edges
- Add proper TRPC endpoints and database queries for new insights
- Follow Cal.com design patterns with consistent styling
- All charts now have individual cards with rounded-2xl borders and shadow-sm elevation
- Fix all linting issues with unused imports and variables

hackice20 committed:
Merge branch 'calcom:main' into feature/insights-noshow-csat-graphs

hackice20 committed:
coderabbit fixes

hackice20 committed:
fix(insights): address review feedback on charts

hackice20 committed:
fix(insights): correct dayjs timezone usage for type safety

hackice20 committed:
Update packages/features/insights/services/InsightsBookingBaseService.ts

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

hackice20 committed:
Merge branch 'main' into feature/insights-noshow-csat-graphs

hackice20 committed:
Fix InfoBadge import path

hackice20 committed:
Wrap InfoBadge in span for styling

hackice20 commented:
@eunjae-lee any updates?

eunjae-lee commented on packages/features/insights/components/booking/BookingKPICards.tsx:
Why did you change this? it seems irrelevant to this PR. Also we'd like to keep the current design for the kpi cards.

<img width="1194" height="730" alt="Screenshot 2025-10-23 at 15 03 13" src="https://github.com/user-attachments/assets/5ca9106c-a6f9-451d-93d0-cd63a579ebf6" />

eunjae-lee reviewed (changes requested):
Thanks for the contribution! I think this is really well done following the existing pattern. I left some feedback that need to be addressed.

hackice20 committed:
Add insights charts and i18n keys

hackice20 commented:
@eunjae-lee 
1) I have moved no_show_hosts_over_time and others to common.json.    

2)
```
+        const bookingDate = new Date(date);
-      const bookingDate = dayjs.tz(date, timeZone).startOf("day").toDate();
  ```
  
  reverted this
  
3) reverted card design  
<img width="1194" height="730" alt="image" src="https://github.com/user-attachments/assets/78492214-7632-4623-ab55-8ce622b47e3e" />

4) used ChartCards 


let me know what you think :)

eunjae-lee commented on packages/features/insights/components/booking/RecentNoShowGuestsChart.tsx:
```diff
@@ -41,42 +41,44 @@ export const RecentNoShowGuestsChart = () => {
       title={t("recent_no_show_guests")}
       titleTooltip={t("recent_no_show_guests_tooltip")}
       className="h-full">
```
Any reason why this component is modified?

eunjae-lee reviewed (changes requested):
Hi @hackice20 I've left more feedback.

hackice20 committed:
Fix ChartCard usage and double card wrapping issues

hackice20 committed:
Merge branch 'main' into feature/insights-noshow-csat-graphs

hackice20 commented:
hopefully this works, let me know if there are any other suggestions or improvements i can make @eunjae-lee :))

eunjae-lee reviewed (changes requested):
Please match the style. I see unnecessary borders.

![Screenshot 2025-10-27 at 16.57.34.png](https://app.graphite.dev/user-attachments/assets/bda168d4-8e42-4713-a005-bfc8a32b0739.png)

![Screenshot 2025-10-27 at 16.58.33.png](https://app.graphite.dev/user-attachments/assets/be3d4784-a1e0-465f-97a4-f1c8c842c59d.png)

hackice20 committed:
Merge branch 'main' into feature/insights-noshow-csat-graphs

hackice20 committed:
Remove double card wrapping for chart components

hackice20 committed:
Merge branch 'feature/insights-noshow-csat-graphs' of https://github.com/hackice20/cal.com into feature/insights-noshow-csat-graphs

eunjae-lee committed:
Merge branch 'main' into feature/insights-noshow-csat-graphs

eunjae-lee reviewed (approved):
Thanks. Looks good!
